### PR TITLE
Doc comment improvements

### PR DESCRIPTION
**Pull request by pepone**
This PR improves the doc comments for generated types

### CODE REVIEW

File: cpp/src/slice2cs/Gen.cpp [hunk 1]
```diff
@@ -2136,31 +2168,32 @@ Slice::Gen::ProxyVisitor::visitInterfaceDefStart(const InterfaceDefPtr& p)
...
+            bool inValue = false;
+
+            _out << sp;
+            _out << nl << "/// <summary>Creates an <see cref=\"ZeroC.Ice.OutgoingRequestFrame\"/> for "
+                 << fixId(operationName(operation)) << " operation.</summary>";
+            _out << nl << "/// <param name=\"proxy\">Proxy to the target Ice Object.</param>";
+            if (paramCount > 0)
             {
-                outputSeparator = true;
+                _out << nl << "/// <param name=\"args\">The remote operation args.</param>";
```
Comment (bernardnormier) [inline cpp/src/slice2cs/Gen.cpp:2187]: I would spell out arguments in the doc.

File: cpp/src/slice2cs/Gen.cpp [hunk 2]
```diff
@@ -2243,28 +2270,48 @@ Slice::Gen::ProxyVisitor::visitInterfaceDefEnd(const InterfaceDefPtr& p)
...
     //
     _out << sp;
+    _out << nl << "/// <summary>Factory for <see cref=\"" << name << "\"/> proxies.</summary>";
     _out << nl << "public static readonly new ZeroC.Ice.ProxyFactory<" << name << "> Factory =";
     _out.inc();
     _out << nl << "(reference) => new _" << p->name() << "Prx(reference);";
     _out.dec();
 
     _out << sp;
+    _out << nl << "/// <summary>An <see cref=\"ZeroC.Ice.InputStreamReader{T}\"/> used to read "
```
Comment (bernardnormier) [inline cpp/src/slice2cs/Gen.cpp:2280]: I would remove the "non nullable". A unqualified proxy type is now non-nullable.

File: cpp/src/slice2cs/Gen.cpp [hunk 3]
```diff
@@ -2662,24 +2704,29 @@ Slice::Gen::DispatcherVisitor::writeReturnValueStruct(const OperationPtr& operat
...
     {
         _out << sp;
+        _out << nl << "/// <summary>Helper struct used to marshal the return value of " << opName << " operation."
+             << "</summary>";
         _out << nl << "public struct " << opName << "MarshaledReturnValue";
         _out << sb;
+        _out << nl << "/// <summary>The frame holding the marshaled response.</summary>";
         _out << nl << "public ZeroC.Ice.OutgoingResponseFrame Response { get; }";
 
+        _out << nl << "/// <summary>Construct a new <see cref=\"" << opName << "MarshaledReturnValue\"/> instance that";
```
Comment (bernardnormier) [inline cpp/src/slice2cs/Gen.cpp:2714]: Constructs

File: cpp/src/slice2cs/Gen.cpp [hunk 4]
```diff
@@ -2662,24 +2704,29 @@ Slice::Gen::DispatcherVisitor::writeReturnValueStruct(const OperationPtr& operat
...
+        _out << nl << "/// immediately marshals the response for the " << opName << "operation.</summary>";
```
Comment (bernardnormier) [inline cpp/src/slice2cs/Gen.cpp:2715]: ... marshals the return value of operation opName.

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: cpp/src/slice2cs/Gen.cpp (hunk 2 above)
Comment (pepone) [reply cpp/src/slice2cs/Gen.cpp:2280]: should we also mark them as `EditorBrowsableState.Never`?

File: cpp/src/slice2cs/Gen.cpp (hunk 4 above)
Comment (pepone) [reply cpp/src/slice2cs/Gen.cpp]: fixed

File: cpp/src/slice2cs/Gen.cpp (hunk 3 above)
Comment (pepone) [reply cpp/src/slice2cs/Gen.cpp]: fixed

File: cpp/src/slice2cs/Gen.cpp (hunk 1 above)
Comment (pepone) [reply cpp/src/slice2cs/Gen.cpp]: fixed

**Pull request merged**